ansible_collections/arista/avd/roles/eos_designs/templates/facts/switch/switch.j2

```diff
@@ -28,11 +28,11 @@ switch:
   mgmt_interface: {{ switch_data.combined.mgmt_interface | arista.avd.default(switch_platform_settings.management_interface, mgmt_interface) }}
 
 {# switch.underlay_routing_protocol #}
-{% if underlay_routing_protocol is arista.avd.defined and underlay_routing_protocol | lower in ['isis', 'ospf'] %}
-  underlay_routing_protocol: {{ underlay_routing_protocol | lower }}
-{% else %}
-  underlay_routing_protocol: 'ebgp'
+{% set switch_underlay_routing_protocol = underlay_routing_protocol | arista.avd.default(switch.default_underlay_routing_protocol) | lower %}
+{% if switch_underlay_routing_protocol not in ['ebgp', 'isis', 'isis-ldp', 'isis-sr', 'isis-sr-ldp', 'ospf', 'ospf-ldp'] %}
+{%     set switch_underlay_routing_protocol = switch.default_underlay_routing_protocol %}
```

Comment:
The logic looks good. But what if the user sets ```default_underlay_routing_protocol``` as **ibgp** or some unknown in node_type_keys. Should we take care of that or no? Any comments @ClausHolbechArista. Because the old logic is simple and even if user provides something unknown, it is converted to **ebgp** because of default nature.

Reply:
My logic was:

```
{%  set switch_underlay_routing_protocol = underlay_routing_protocol | arista.avd.default(switch.default_underlay_routing_protocol) | lower %}
{%  if switch_underlay_routing_protocol in [isis, isis-ldp, isis-sr, isis-sr-ldp, ospf, ospf-ldp] %}
underlay_routing_protocol: {{ switch_underlay_routing_protocol }}
{%  else %}
underlay_routing_protocol: ebgp
{% endif %}
```
Also we can remove the **ebgp** as default in ```default_underlay_routing_protocol```

Reply:
Not needed. The `node_type_keys` and hence the `default_underlay_routing_protocol` is only used by "experts" anyway, so I don't think we need to add extra protection there. It will be sorted with schemas later.